### PR TITLE
Add debug option: ignoresysdns v2

### PR DESCRIPTION
**Pull request by licaon-kter**
Simplified it

### CODE REVIEW

File: app/src/main/java/eu/faircode/netguard/Util.java [hunk 1]
```diff
@@ -279,15 +279,23 @@ public static boolean hasPhoneStatePermission(Context context) {
...
+        String vpnDns = prefs.getString("dns", null);
+        boolean ignoresysdns = prefs.getBoolean("ignoresysdns", false);
         List<String> listDns = new ArrayList<>();
-        if (!TextUtils.isEmpty(dns1))
-            listDns.add(dns1);
-        if (!TextUtils.isEmpty(dns2))
-            listDns.add(dns2);
-        if (TextUtils.isEmpty(dns1) && TextUtils.isEmpty(dns2))
-            listDns.add("8.8.8.8");
+        if (!ignoresysdns) {
```
Comment (M66B) [inline app/src/main/java/eu/faircode/netguard/Util.java]: It is best practice not to negate condition when there is no need to (when there is also an else).

**Comment by licaon-kter** [comment]
So if VPN DNS is empty you end up DNSless?
VPN DNS value in settings will be empty, also no hint will be shown.

**Comment by M66B** [comment]
Ignore is ignore, so you'll be responsible to provide a DNS server yourself.

**Comment by licaon-kter** [comment]
:)

**Comment by M66B** [comment]
I have revert the change, since it will let the settings crash

**Comment by licaon-kter** [comment]
`use_defdns` in XML vs `usedefdns` in .java might have been at fault?

**Comment by M66B** [comment]
The settings assume there is at least one DNS server.

**Comment by M66B** [comment]
Can you tell me why you want this setting again?

**Pull request closed without merging**